### PR TITLE
We can't add a volume by Key but volume object directly

### PR DESCRIPTION
**Pull request by gnufied**
We should add PV object directly.

cc @jsafrane @RaunakShah  @wongma7

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -1439,7 +1439,7 @@ func (ctrl *ProvisionController) provisionClaimOperation(ctx context.Context, cl
 	if err := ctrl.volumeStore.StoreVolume(claim, volume); err != nil {
 		return ProvisioningFinished, err
 	}
-	if err = ctrl.volumes.Add(volume.Name); err != nil {
+	if err = ctrl.volumes.Add(volume); err != nil {
```
Comment (RaunakShah) [inline controller/controller.go:1442]: What about changing this too? https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/controller/controller.go#L1520

**Comment by RaunakShah** [comment]
@gnufied are you planning to merge this PR?

**Comment by gnufied** [comment]
@RaunakShah updated.

**Comment by wongma7** [comment]
Hmm the test is failing but not in identical PR https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/pull/92/files

/retest

**Comment by wongma7** [comment]
Dunno if the tests are racy now, I don't remember this being a restriction:

```

W1009 19:38:48.268298   11788 shared_informer.go:488] resyncPeriod 100000000 is too small. Changing it to the minimum allowed value of 1000000000 
```

/retest

**Pull request merged**